jax/experimental/jax2tf/shape_poly.py

```diff
@@ -1186,31 +1227,71 @@ def eval_operand(o: DimSize) -> Union[int, jax.Array]:
         if not (left_int >= right_int):
           raise ValueError(self.make_err_msg(left_int, right_int))
       else: assert False
-    else:
-      return None  # TODO: evaluate constraint dynamically
+      return None
+
+    if self.comp == ShapeConstraint.Comparator.EQ:
+      is_ok = lax.eq(left, right)
```

Comment:
Question: why line 1190 use overload "==" operator here we use `lax.eq` primitive ? I assume they are doing the same job ?